docs/contracts/node-operators-registry.md

```diff
@@ -116,46 +95,489 @@ function getOperatorIds()
+
+### removeOperator()
+
+Removes the node operator with given id
+
+```sol
+function removeOperator(uint256 _operatorId)
+        external
         override
-        returns (bool[] memory)
 ```
 
+#### Parameters:
+
+| Name      | Type      | Description                       |
+| --------- | --------- | --------------------------------- |
+| `_id`     | `uint256` | Node Operator id                  |
+
+### joinOperator()
+
+Adds a validator that was already staked on the polygon stake manager to the PoLido protocol.
+
+```sol
+function joinOperator() external override
+```
+
+### stake()
+
+Stakes a validator on the Polygon stakeManager contract.
+
+```sol
+function stake(uint256 _amount, uint256 _heimdallFee)
+```
+
+#### Parameters:
+
+| Name    | Type      | Description         |
+| ------- | --------- | ------------------- |
+| `_amount`   | `uint256` | Amount to stake   |
+| `_heimdallFee` | `uint256`  | Heimdall fee |
+
+### restake()
+
+Restakes Matics to the validator of corresponding owner on the Polygon stakeManager
+
+```sol
+function restake(uint256 _amount, bool _restakeRewards)
+        external
+        override
+```
+
+#### Parameters:
+
+| Name    | Type      | Description         |
+| ------- | --------- | ------------------- |
+| `_amount`   | `uint256` | Amount to stake   |
+| `_restakeRewards` | `bool`  | If true, the rewards will also be restaked |
+
+### unstake()
+
+Unstakes a validator from the Polygon stakeManager contract.
+
+```sol
+function unstake() external override
+```
+
+### unjail()
+
+Unjails the validator and turns its status from UNSTAKED to ACTIVE.
+
+```sol
+function unjail() external override
+```
+
+### migrate()
+
+Migrates the validator ownership to reward address
+
+```sol
+function migrate() external override
+```
+
+### topUpFee()
+
+Tops up heimdall fees.
+
+```sol
+function topUpForFee(uint256 _heimdallFee)
+        external
+        override
+```
+
+#### Parameters:
+
+| Name    | Type      | Description         |
+| ------- | --------- | ------------------- |
+| `_heimdallFee`   | `uint256` | Amount of Matic that will be added to the current heimdallFee provided   |
+
+### unstakeClaim()
+
+Begins the unstaking process of the staked tokens. Tokens will be unstaked after the withdraw delay has passed.
+
+```sol
+function unstakeClaim() external override
+```
+
+### claimFee()
+
+Withdraws the Heimdall fees.
+
+```sol
+function claimFee(
+        uint256 _accumFeeAmount,
+        uint256 _index,
+        bytes memory _proof
+    ) external override
+```
+#### Parameters:
+
+| Name    | Type      | Description         |
+| ------- | --------- | ------------------- |
+| `_accumFeeAmount`   | `uint256` | Amount of Heimdall fees in Matic that will be withdrawn   |
+| `_index`   | `uint256` | Index of the Validator   |
+| `_proof`   | `bytes` | Proof for the Stake manager  |
+
+### withdrawRewards()
+
+Withdraws rewards to the operator owner.
```

Comment:
Withdraws staking rewards accumulated by a validator and transfers the amount to the  to the operator's owner.